galsim/config/noise.py

```diff
@@ -457,19 +457,20 @@ def addNoise(self, config, base, im, rng, current_var, draw_method, logger):
         # Get how much extra sky to assume from the image.noise attribute.
         sky = GetSky(base['image'], base, logger, full=True)
         extra_sky = GetSky(config, base, logger, full=True)
-        total_sky = sky + extra_sky # for the return value
+        total_sky = sky + extra_sky
         if isinstance(total_sky, Image):
-            var = np.mean(total_sky.array) + read_noise_var
+            var_adu = np.mean(total_sky.array) / gain + read_noise_var / gain**2
         else:
-            var = total_sky + read_noise_var
+            var_adu = total_sky / gain + read_noise_var / gain**2
```

Comment:
This is now the same as `target_var` below, so you can remove that name and use `var_adu` there.